src/NHapi.Base/Parser/PipeParser.cs

```diff
@@ -406,8 +406,7 @@ public virtual void Parse(ISegment destination, string segment, EncodingCharacte
         public override void Parse(IMessage message, string @string, ParserConfiguration parserConfiguration)
         {
             var structureDefinition = GetStructureDefinition(message);
```

Comment:
We probably need a null guard here to protect against passing null (protecting the consumer from themselves 😂)

```csharp
if(parserConfiguration is null)
{
    throw new ArgumentNullException(nameof(parserConfiguration));
}
```